<launch>

<group ns="see3cam" >
  <node pkg="uvc_camera" type="uvc_camera_node" name="uvc_camera" output="screen">
    <param name="width" type="int" value="1280" /> 
    <!-- we raised the value by the factor 2, as it is supported by previous output -->
    <param name="height" type="int" value="720" /> 
    <!-- we raised the value by the factor 2 -->
    <param name="fps" type="int" value="60" />
    <param name="frame" type="string" value="wide_stereo" />
       
    <param name="device" type="string" value="/dev/video0" /> 
    <param name="camera_info_url" type="string" value="file://$(find uvc_camera)/example.yaml" />

            <remap from="image_raw" to="camera/image_raw" />
            <remap from="camera_info" to="camera/camera_info" />

  </node>

    <!-- Launches the node that will rectify the image based on the calibration .yaml file. -->

    <node ns="camera" pkg="image_proc" type="image_proc" name="image_proc" if="0"/>
    
    <group>

        <!-- node to reduce the size of raw camera image -->
        <node pkg="nodelet" type="nodelet" 
            args="standalone image_proc/crop_decimate" 
            name="decimator">

            <param name="decimation_x" type="int" value="2" />
            <param name="decimation_y" type="int" value="2" />
    
        </node>
    </group>    
  </group>
</launch>
